wrapanapi/containers/__init__.py

```diff
@@ -1,179 +0,0 @@
-    LabelNotFoundException, ResourceAlreadyExistsException,
-    UncreatableResourceException)
-
-
-class ContainersResourceBase(object):
-    """
-    A container resource base class. This class includes the base functions of (almost)
-    all container resources. Each resource has its own API entry and use different API
-    (Kubernetes or OpenShift). Each resource has get, post, patch and delete methods which
-    directed to the path of the resource.
-    The following parameters should be statically defined:
-        * RESOURCE_TYPE: (str) the resource type name in the API
-        * CREATABLE (optional): (bool) Specify whether this resource is creatable or not (some
-                                resources are not, e.g. Pod is created by Replication Controller
-                                and not manually). set to False by default.
-        * (optional) API: (str) The API to use - the default is Kubernetes ('k_api') but some
-                          resources use OpenShift ('o_api').
-        * (optional) VALID_NAME_PATTERN: (str) the regex pattern that match a valid object name
-        * (optional) KIND: (str) the resource 'Kind' property as it appear in JSON.
-                           if not specified, grabbing the Kind from the class name.
-    """
-    CREATABLE = False
-    API = 'k_api'
-
-    def __init__(self, provider, name, namespace):
-        """
-        Args:
-            provider: (Openshift || Kubernetes) The containers provider
-            name: (str) The name of the resource
-            namespace: (str) The namespace used for this resource
-        """
-        if hasattr(self, 'VALID_NAME_PATTERN') and not re.match(self.VALID_NAME_PATTERN, name):
-            raise InvalidValueException('{0} name "{1}" is invalid. {0} name must '
-                                        'match the regex "{2}"'
-                                        .format(self.RESOURCE_TYPE, name, self.VALID_NAME_PATTERN))
-        self.provider = provider
-        self.name = name
-        self.namespace = namespace
-
-    def __eq__(self, other):
-        return (getattr(self, 'namespace', None) == getattr(other, 'namespace', None) and
-                self.name == getattr(other, 'name', None))
-
-    def __repr__(self):
-        return '<{} name="{}" namespace="{}">'.format(
-            self.__class__.__name__, self.name, self.namespace)
-
-    def exists(self):
-        """Return whether this object exists or not."""
-        try:
-            self.get()
-            return True
-        except RequestFailedException:
-            return False
-
-    @cached_property
-    def api(self):
-        """Return the used API according to the defined API."""
-        return getattr(self.provider, self.API)
-
-    @classmethod
-    def kind(cls):
-        """Return the resource Kind property as it should be in the JSON"""
-        return getattr(cls, 'KIND', None) or cls.__name__
-
-    @classmethod
-    def create(cls, provider, payload):
```

Comment:
I dont see any replacement for this

Reply:
all this stuff is available thru o_api. I can add missing create_ or patch_ methods if those are really used somewhere.

Reply:
They are not used at the moment but I would like to start updating our tests to use them. Right now, we grab an object and perform various actions on it. I would like to change that to: create an object, perform actions, remove object.

Reply:
it won't be difficult to add missing methods. I'll help you.
Let me know when you decide to do that. I think we will restructure module.

Reply:
Okay, I agree with this.